packages/frontend/src/components/Setup/FormStep/index.test.jsx

```diff
@@ -0,0 +1,66 @@
+import { setupTests } from 'helpers/tests';
+import FormStep from '.';
+import { routes } from '../../../utils';
+import { baseState } from 'contexts';
+import { produce } from 'immer';
+
+describe('FormStep component', () => {
+  it('renders the correct label and form element at step one', () => {
+    const { label, placeholder, buttonText, buttonColorClass, FormElement } =
+      routes[0].children[0].elementProps;
+    const { name, schema } = routes[0].children[0];
+
+    const props = {
+      label,
+      placeholder,
+      buttonText,
+      buttonColorClass,
+      name,
+      schema,
+      FormElement,
+    };
+    const state = { ...baseState };
+
+    const { screen } = setupTests(FormStep, { props, state });
+
+    const stepOneComponent = screen.getByText("Enter the first player's name");
+
+    expect(stepOneComponent).toBeInTheDocument();
+
+    expect(screen.queryByText("Enter the second player's name")).toBeNull();
+
+    const input = document.querySelector('input[type="name"]');
+    expect(input).toBeInTheDocument();
+  }),
+    it('renders the correct label and form element at step three', () => {
+      const { label, placeholder, buttonText, buttonColorClass, FormElement } =
+        routes[0].children[2].elementProps;
+      const { name, schema } = routes[0].children[2];
```

Comment:
I feel like we should have a helper function for this, since it's used in multiple places. If we added a prop to the element, we'd have to go back through and add it to all the various tests where it's used.

The helper could be something like `getFormStepPropsFromRouteChild` (long name, I know, but at least it's descriptive?). The function could define an array of desired keys from the elementProps object, filter the keys based on that array and then use Array.reduce to reconstruct the object, which would then be returned.

Reply:
This is really cool. I think I did this correctly and it was an awesome learning experience because I haven't used Object.forEntries or Object.entries IRL before.